### ISSUE TITLE
allow users to copy/paste text

### ISSUE THREAD

**Issue by keianhzo** [MozillaReality/FirefoxReality#189]
When selecting a text and long pressing it an actions menu should show up with actions such as copy, share, select all, Google Search, etc

This applies to:
- Navigation Bar
- Browser window

**Comment by avrignaud** [comment]
Thanks, let's add @thenadj so we can review. In the future we are looking
to have a "Design" tag as well for tracking issues.

**Comment by thenadj** [comment]
Yes, we need to add this feature. I recommend making it a priority for v1.1

**Comment by avrignaud** [comment]
Question: *do* we have any long-tapping actions wired up in the browser right now? We don't have search, send to device... is there any native GeckoView support we should be aware of that would cause us to try and do this for 1.0? Otherwise I'm strongly supportive of it being a 1.1 feature.

**Comment by bluemarvin** [comment]
GeckoView supports long pressing on content. Currently, as-is, the text that is long pressed is highlighted and drag handles appear. But since there is no copy/cut/paste action menu, there isn't much you can do with the selected text. GeckoView does have an api to get the selected text once we have UX design and UI implemented for the long press actions. Long pressing on a link does nothing right now, but GeckoView provides a callback so we can present the correct UI.

**Comment by cvan** [comment]
We will want to also revisit this for context menus in web-page content, for example for "right-clicking" (or long-pressing) to allow Open in a New Tab, etc.

**Comment by cvan** [comment]
This is probably the most common Browser request I read about from users in the OculusGo sub-Reddit. Latest thread: https://www.reddit.com/r/OculusGo/comments/99pxej/copypaste_in_browser_wasnt_even_on_the_horizon/

**Comment by avrignaud** [comment]
Thanks @cvan ! We'll definitely look at adding copy/paste support to long presses in 1.1. Easy sharing to social media and other services is also high on the list.

**Comment by BioeJD** [comment]
Why is this pushed all the way back to v1.4? That's a critical feature with much higher priority than most of the other things I see on earlier milestones. It's already a complete pain to type anything in VR, so no copy-paste ability makes certain actions essentially impossible.

**Comment by bluemarvin** [comment]
Will need https://bugzilla.mozilla.org/show_bug.cgi?id=1570810 in order to correctly implement back button functionality with copy cut paste.

**Comment by jvonitter** [comment]
Design is complete https://trello.com/c/AEOHvjHr/427-uis-85-copy-paste

**Comment by jvonitter** [comment]
Spec has been updated. The change is to the placement of the long-press menu location. Instead of being centered in the middle of the page (like a dialog), its placement and behavior should be more like the long-press menu for regular text.
https://trello.com/c/AEOHvjHr/427-uis-85-copy-paste

**Comment by jvonitter** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1570810 is done. no longer blocked!

**Comment by jvonitter** [comment]
Design has been updated to match implementation: https://trello.com/c/AEOHvjHr/427-uis-85-copy-paste

Also specifies hover behavior for CCP menu.

**Comment by philip-lamb** [comment]
> https://bugzilla.mozilla.org/show_bug.cgi?id=1570810 is done. no longer blocked!

It was closed as `wontfix`, with the recommendation to wait on the fix in https://bugzilla.mozilla.org/show_bug.cgi?id=1581161.

**Comment by bluemarvin** [comment]
PR #1883 is not currently blocked by this issue.
